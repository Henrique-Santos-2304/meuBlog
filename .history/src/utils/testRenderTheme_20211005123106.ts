import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
