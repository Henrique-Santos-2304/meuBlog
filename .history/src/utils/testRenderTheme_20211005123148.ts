import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
