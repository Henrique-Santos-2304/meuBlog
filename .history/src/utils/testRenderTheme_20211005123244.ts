import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
