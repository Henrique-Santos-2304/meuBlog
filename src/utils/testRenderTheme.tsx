import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
