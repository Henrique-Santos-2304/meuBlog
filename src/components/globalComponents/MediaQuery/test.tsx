import HandleMediaQuery from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<MediaQuery />", () => {
  it("should render in specified breakpoint with show is true ", () => {
    const { container } = renderTheme(
      <HandleMediaQuery responsive="max-width" asWidth="767px" show>
        <p>Media Query</p>
      </HandleMediaQuery>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId(/media-query/i)).toHaveStyleRule(
      "display",
      "block",
      {
        media: "(max-width:767px)",
      }
    );
  });
  it("should not render in specified breakpoint with show is false ", () => {
    renderTheme(
      <HandleMediaQuery responsive="max-width" asWidth="767px" show={false}>
        <p>Media Query</p>
      </HandleMediaQuery>
    );

    expect(screen.queryByTestId(/media-query/i)).toHaveStyleRule(
      "display",
      "none",
      {
        media: "(max-width:767px)",
      }
    );
  });
});
