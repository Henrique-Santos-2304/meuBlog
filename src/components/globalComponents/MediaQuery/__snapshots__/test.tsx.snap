// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<MediaQuery /> should render in specified breakpoint with show is true  1`
] = `
.c0 {
  display: none;
}

@media (max-width:767px) {
  .c0 {
    display: block;
  }
}

<div>
  <div
    class="c0"
    data-testid="media-query"
  >
    <p>
      Media Query
    </p>
  </div>
</div>
`;
